apps/public/src/app/(general)/bijlage/Attachments.tsx

```diff
@@ -10,36 +10,52 @@ import { postMeldingByMeldingIdAttachment } from 'apps/public/src/apiClientProxy
 
 import { BackLink } from '../_components/BackLink'
 
-import { FileUpload } from './_components/FileUpload'
+import { FileInput } from './_components/FileInput'
+import { FileList } from './_components/FileList'
 import { redirectToNextPage } from './actions'
 
+const MAX_FILES = 3
+
 type Props = {
   meldingId: number
   token: string
 }
 
-export const Bijlage = ({ meldingId, token }: Props) => {
-  const [uploadedFiles, setUploadedFiles] = useState<AttachmentOutput[]>([])
+export type UploadedFiles = AttachmentOutput & { image: string }
+
+export const Attachments = ({ meldingId, token }: Props) => {
+  const [uploadedFiles, setUploadedFiles] = useState<UploadedFiles[]>([])
   const [errorMessage, setErrorMessage] = useState<string>()
 
   const handleOnChange = async (event: ChangeEvent<HTMLInputElement>) => {
+    setErrorMessage(undefined)
+
     if (!event.currentTarget.files) return
 
     const files = Array.from(event.currentTarget.files)
 
+    if (files.length > MAX_FILES) {
+      setErrorMessage(`Je kunt maximaal ${MAX_FILES} bestanden uploaden.`)
+      return
+    }
+
     try {
       const result = await Promise.all(
-        files.map((file) =>
-          postMeldingByMeldingIdAttachment({
+        files.map(async (file) => {
+          const uploadedFile = await postMeldingByMeldingIdAttachment({
             formData: { file },
             meldingId,
             token,
-          }),
-        ),
+          })
+
+          const image = URL.createObjectURL(file)
+
+          return { ...uploadedFile, image }
+        }),
       )
       setUploadedFiles((currentFiles) => [...currentFiles, ...result])
     } catch (error) {
-      setErrorMessage((error as Error).message)
+      setErrorMessage((error as ApiError).message)
```

Comment:
This doesn't really work when trying to upload 2 files where 1 has the wrong extension.

Reply:
This is fixed when using the accept prop right?

Reply:
Is that expected behaviour?

Reply:
Creating a separate ticket as discussed